edk2toolext/capsule/pyopenssl_signer.py

```diff
@@ -21,10 +22,30 @@ def sign(data, signature_options, signer_options):
     dictionaries that are used by capsule_tool and capsule_helper
     '''
     # NOTE: Currently, we only support the necessary algorithms for capsules.
-    if signature_options['sign_alg'] != 'pkcs12':
-        raise ValueError(f"Unsupported signature algorithm: {signature_options['sign_alg']}")
+
+    # The following _if_ clause handles the deprecated signature_option 'sign_alg' for backwards compatibility
+    # when the deprecated option is supplied, this code adds the new, required options based on prior code behavior
+    if 'sign_alg' in signature_options:
+        warnings.warn('Signature_option "sign_alg" is deprecated, use "type"', DeprecationWarning)
+        if signature_options['sign_alg'] == 'pkcs12':
+            # map legacy behavior to new options and backwards-compatible values
+            signature_options['type'] = 'bare'
+            signature_options['encoding'] = 'binary'
+            signer_options['key_file_format'] = 'pkcs12'
+        else:
+            raise ValueError(f"Unsupported signature algorithm: {signature_options['sign_alg']}!")
+
+    ''' signature type 'bare' is just a signed digest, no headers/footers or ASN '''
+    if signature_options['type'] != 'bare':
+        raise ValueError(f"Unsupported signature type: {signature_options['type']}")
+    if 'type_options' in signature_options:
+        raise ValueError(f"Signature type options not supported")
+    if signature_options['encoding'] != 'binary':
```

Comment:
Binary in this case is PEM?

Reply:
As I understand it, PEM is base64 with headers and footers.
This is the raw, binary signed digest... followed by 0x0A... no other headers or footers.

Reply:
Scratch to 0x0A, that was being added by an editor.
See line 38 docstring

Reply:
Resolved in latest commit